From 6933349753658b6337d15629cf5a41ebcbf82ec6 Mon Sep 17 00:00:00 2001
From: Ryan Gonzalez <rymg19@gmail.com>
Date: Tue, 24 Jan 2023 02:02:21 +0000
Subject: [PATCH] Preload the Asahi drivers when preparing the GPU sandbox

This replaces 4119913's changes and instead uses the existing dlopen
calls for kmsro drivers.

Bug: None
Change-Id: I3206f516dddd1f05715484a579bdef34ab0b36dd
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4186738
Reviewed-by: Matthew Denton <mpdenton@chromium.org>
Commit-Queue: Matthew Denton <mpdenton@chromium.org>
Reviewed-by: Kenneth Russell <kbr@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1095970}
---
 content/gpu/gpu_sandbox_hook_linux.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/content/gpu/gpu_sandbox_hook_linux.cc b/content/gpu/gpu_sandbox_hook_linux.cc
index 0ae80a2fb3dcd..3202de2f37a5d 100644
--- a/content/gpu/gpu_sandbox_hook_linux.cc
+++ b/content/gpu/gpu_sandbox_hook_linux.cc
@@ -471,15 +471,18 @@ void LoadArmGpuLibraries() {
         DRI_DRIVER_DIR "/panfrost_dri.so",
         DRI_DRIVER_DIR "/mediatek_dri.so",
         DRI_DRIVER_DIR "/rockchip_dri.so",
+        DRI_DRIVER_DIR "/asahi_dri.so",
 #else
         "/usr/lib64/dri/msm_dri.so",
         "/usr/lib64/dri/panfrost_dri.so",
         "/usr/lib64/dri/mediatek_dri.so",
         "/usr/lib64/dri/rockchip_dri.so",
+        "/usr/lib64/dri/asahi_dri.so",
         "/usr/lib/dri/msm_dri.so",
         "/usr/lib/dri/panfrost_dri.so",
         "/usr/lib/dri/mediatek_dri.so",
         "/usr/lib/dri/rockchip_dri.so",
+        "/usr/lib/dri/asahi_dri.so",
 #endif
         nullptr
       };
-- 
2.39.0

